### PR TITLE
Elevator backup summaries

### PR DESCRIPTION
**Pull request by digitalcora**
Best reviewed per-commit as there are some chonky changes prior to the "main" one. Commit messages have the details on each change.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208677840815949

### CODE REVIEW

File: lib/screens/v2/candidate_generator/elevator/closures.ex [hunk 1]
```diff
@@ -24,164 +43,145 @@ defmodule Screens.V2.CandidateGenerator.Elevator.Closures do
...
-      (Alert.informs_stop_id?(alert, home_stop_id) or not has_nearby_redundancy?(alert))
-  end
+  defp elevator_closure(%Alert{id: id, effect: :elevator_closure, informed_entities: entities}) do
+    # We expect there is a 1:1 relationship between `elevator_closure` alerts and individual
+    # out-of-service elevators. Log a warning if our assumptions don't hold.
+    stations_and_facilities =
+      entities
+      |> Enum.filter(&(InformedEntity.parent_station?(&1) and not is_nil(&1.facility)))
+      |> Enum.map(fn %{facility: facility, stop: station_id} -> {station_id, facility} end)
+      |> Enum.uniq()
```
Comment (robbie-sundstrom) [inline lib/screens/v2/candidate_generator/elevator/closures.ex:82]: Appreciate how readable this is!

File: lib/screens/v2/candidate_generator/elevator/closures.ex [hunk 2]
```diff
@@ -24,164 +43,145 @@ defmodule Screens.V2.CandidateGenerator.Elevator.Closures do
...
+  # Elevators at the home station ID are always relevant.
+  defp relevant_closure?(%Closure{station_id: station_id}, station_id, _closures), do: true
+
+  # If any of a closed elevator's alternates are also closed, it's always relevant.
+  defp relevant_closure?(
+         %Closure{elevator: %Elevator{alternate_ids: alternate_ids, redundancy: redundancy}},
+         _home_station_id,
+         closures
+       ) do
+    Enum.any?(closures, fn %Closure{id: id} -> id in alternate_ids end) or redundancy != :nearby
```
Comment (robbie-sundstrom) [inline lib/screens/v2/candidate_generator/elevator/closures.ex:165]: appreciate these comments

File: test/screens/v2/candidate_generator/elevator/closures_test.exs [hunk 3]
```diff
@@ -229,28 +213,31 @@ defmodule Screens.V2.CandidateGenerator.Elevator.ClosuresTest do
...
+                   ^footer_instance
+                 ] =
+                   ElevatorClosures.elevator_status_instances(
+                     struct(Screen, app_id: :elevator_v2, app_params: config),
+                     header_instance,
+                     footer_instance
+                   )
+        end)
+
+      assert logs =~ "elevator_closure_affects_multiple"
```
Comment (robbie-sundstrom) [inline test/screens/v2/candidate_generator/elevator/closures_test.exs:233]: nice testing of this edge case

File: lib/screens/v2/candidate_generator/elevator/closures.ex [hunk 4]
```diff
@@ -1,8 +1,27 @@
 defmodule Screens.V2.CandidateGenerator.Elevator.Closures do
   @moduledoc false
 
+  defmodule Closure do
```
Comment (robbie-sundstrom) [inline lib/screens/v2/candidate_generator/elevator/closures.ex:4]: Nit: Most of our modules don't have docs, but I wouldn't mind if we started adding them more. Not necessary to merge

**Review by robbie-sundstrom — APPROVED** [review]
Overall looks great! Thanks for running through this last week.

File: lib/screens/v2/candidate_generator/elevator/closures.ex (hunk 4 above)
Comment (digitalcora) [reply lib/screens/v2/candidate_generator/elevator/closures.ex:4]: Good call. Conceptually this one is a "private module" (non-reusable implementation detail of its parent module), and I would like to keep the `@moduledoc false` as a signal of that, but it could have a non-`moduledoc` comment explaining its purpose better.
